tests/fdb/test_fdb.py

```diff
@@ -224,20 +266,41 @@ def test_fdb(ansible_adhoc, ptfadapter, duthosts, rand_one_dut_hostname, ptfhost
             available_ports_idx.append(idx)
 
     vlan_table = {}
+    config_portchannels = conf_facts.get('PORTCHANNEL', {})
 
     for name, vlan in conf_facts['VLAN'].items():
         vlan_table[name] = []
-        ifnames = conf_facts['VLAN_MEMBER'][name].keys()
-        vlan_table[name] = [ conf_facts['port_index_map'][ifname] for ifname in ifnames
-        if conf_facts['port_index_map'][ifname] in available_ports_idx ]
 
-    vlan_member_count = sum([ len(members) for name, members in vlan_table.items() ])
+        for ifname in conf_facts['VLAN_MEMBER'][name].keys():
+            if 'tagging_mode' not in conf_facts['VLAN_MEMBER'][name][ifname]:
+                continue
+            tagging_mode = conf_facts['VLAN_MEMBER'][name][ifname]['tagging_mode']
+            port_index = []
+            if ifname in config_portchannels:
+                for member in config_portchannels[ifname]['members']:
+                    if conf_facts['port_index_map'][member] in available_ports_idx:
+                        port_index.append(conf_facts['port_index_map'][member])
+            elif conf_facts['port_index_map'][ifname] in available_ports_idx:
+                    port_index.append(conf_facts['port_index_map'][ifname])
+            if port_index:
+                vlan_table[name].append({'port_index':port_index, 'tagging_mode':tagging_mode})
+
+    vlan_member_count = sum([ len(members) for members in vlan_table.values() ])
 
     fdb = setup_fdb(ptfadapter, vlan_table, router_mac, pkt_type)
     for vlan in vlan_table:
+        vlan_id = int(vlan.replace('Vlan', ''))
         for src, dst in itertools.combinations(vlan_table[vlan], 2):
-            for src_mac, dst_mac in itertools.product(fdb[src], fdb[dst]):
-                send_recv_eth(ptfadapter, src, src_mac, dst, dst_mac)
+            if 'port_index' not in src or 'tagging_mode' not in src:
+                continue
+            if 'port_index' not in dst or 'tagging_mode' not in dst:
+                continue
+            src_vlan = vlan_id if src['tagging_mode'] == 'tagged' else 0
+            dst_vlan = vlan_id if dst['tagging_mode'] == 'tagged' else 0
+            src_ports = src['port_index']
+            dst_ports = dst['port_index']
+            for src_mac, dst_mac in itertools.product(fdb[src_ports[0]], fdb[dst_ports[0]]):
+                send_recv_eth(ptfadapter, src_ports, src_mac, dst_ports, dst_mac, src_vlan, dst_vlan)
 
     # Should we have fdb_facts ansible module for this test?
     res = duthost.command('show mac')
```

Comment:
We can further check the ifname, it should be a known port or port channel, also a direct member of that vlan (not a port in one portchannel in vlan).

Previously we have a bug, ifname shows a hex string, it could pass the current test case.